airbyte-bootloader/src/main/java/io/airbyte/bootloader/Bootloader.java

```diff
@@ -0,0 +1,291 @@
+import io.airbyte.db.instance.DatabaseMigrator;
+import io.airbyte.persistence.job.JobPersistence;
+import io.airbyte.validation.json.JsonValidationException;
+import io.micronaut.context.annotation.Value;
+import jakarta.annotation.PostConstruct;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Ensures that the databases are migrated to the appropriate level.
+ */
+@Singleton
+public class Bootloader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Bootloader.class);
+  private static final AirbyteVersion VERSION_BREAK = new AirbyteVersion("0.32.0-alpha");
+
+  private final boolean autoUpgradeConnectors;
+  private final ConfigRepository configRepository;
+  private final DatabaseMigrator configsDatabaseMigrator;
+  private final DatabaseInitializer configsDatabaseInitializer;
+  private final AirbyteVersion currentAirbyteVersion;
+  private final Optional<DefinitionsProvider> definitionsProvider;
+  private final FeatureFlags featureFlags;
+  private final DatabaseInitializer jobsDatabaseInitializer;
+  private final DatabaseMigrator jobsDatabaseMigrator;
+  private final JobPersistence jobPersistence;
+  private Runnable postLoadExecution;
+  private final ProtocolVersionChecker protocolVersionChecker;
+  private final boolean runMigrationOnStartup;
+  private final SecretMigrator secretMigrator;
+
+  public Bootloader(
+                    @Value("${airbyte.protocol.auto-upgrade-connectors}") final boolean autoUpgradeConnectors,
+                    final ConfigRepository configRepository,
+                    @Named("configsDatabaseInitializer") final DatabaseInitializer configsDatabaseInitializer,
+                    @Named("configsDatabaseMigrator") final DatabaseMigrator configsDatabaseMigrator,
+                    final AirbyteVersion currentAirbyteVersion,
+                    final Optional<DefinitionsProvider> definitionsProvider,
+                    final FeatureFlags featureFlags,
+                    @Named("jobsDatabaseInitializer") final DatabaseInitializer jobsDatabaseInitializer,
+                    @Named("jobsDatabaseMigrator") final DatabaseMigrator jobsDatabaseMigrator,
+                    final JobPersistence jobPersistence,
+                    final ProtocolVersionChecker protocolVersionChecker,
+                    @Value("${airbyte.run-migration-on-startup}") final boolean runMigrationOnStartup,
+                    final SecretMigrator secretMigrator) {
+    this(autoUpgradeConnectors, configRepository, configsDatabaseInitializer, configsDatabaseMigrator, currentAirbyteVersion,
+        definitionsProvider, featureFlags, jobsDatabaseInitializer, jobsDatabaseMigrator, jobPersistence, protocolVersionChecker,
+        runMigrationOnStartup, secretMigrator, null);
+  }
+
+  @VisibleForTesting
```

Comment:
I think this needs to be public, the `postLoadExecution` is overridden in cloud.